torch_ort_inference/torch_ort_inference/ortinferencemodule/ortinferencemodule.py

```diff
@@ -0,0 +1,255 @@
+           self._inference_session.run_with_iobinding(io_binding, run_options)
+           forward_outputs = io_binding.get_outputs()
+
+           # forward outputs is a list (std::vector<OrtValue>) but _ortvalues_to_torch_tensor
+           # is expected a OrtValueVector (also std::vector<OrtValue> but defined in onnxruntime-training).
+           # _ortvalues_to_torch_tensor_list needs to be used.
+           user_outputs = _utils._ortvalues_to_torch_tensor_list(forward_outputs, self._device)
+           result = [(output.shape, output.device, output.dtype) for output in user_outputs]
+           return _io.unflatten_user_output(self._module_output_schema, result)
+
+        except Exception as e:
+            # Catch-all
+            raise e
+
+    def _export_model(self, input_schema, *inputs, **kwargs):
+        """Exports PyTorch `self._flattened_module` to ONNX for inferencing or training, using `*inputs` and `**kwargs` as input
+        TODO: How to support dynamic axes? Dimensions are determined by samples
+        """
+
+        # Setup dynamic axes for onnx model
+        self._input_info = _io.parse_inputs_for_onnx_export(self._module_parameters, None, input_schema, inputs, kwargs)
+        (
+            output_names,
+            output_dynamic_axes,
+            self._module_output_schema,
+        ) = _io.parse_outputs_for_onnx_export_and_extract_schema(self._original_module, inputs, kwargs)
+        self._input_info.dynamic_axes.update(output_dynamic_axes)
+
+        # FlattenedModule needs _InputInfo to expand user input from *args to
+        # *args + **kwargs
+        self._flattened_module._input_info = self._input_info
+
+        # Export torch.nn.Module to ONNX
+        f = io.BytesIO()
+
+        # Deepcopy inputs, since input values may change after model run.
+        # NOTE: Inputs may contain tensors that have attributes preventing their deepcopy (example grad_fn).
+        # Therefore, deepcopy only the data component of the input tensors for
+        # export.
+        sample_inputs_copy, sample_kwargs_copy = _io.deepcopy_model_input(*inputs, **kwargs)
+        # NOTE: Flattening the input will change the 'input schema', resulting
+        # in a re-export
+        sample_inputs_as_tuple = tuple(self._input_info.flatten(sample_inputs_copy, sample_kwargs_copy, self._device))
+        # Ops behaving differently under train/eval mode need to exported with the
+        # correct training flag to reflect the expected behavior.
+        # For example, the Dropout node in a model is dropped under eval mode.
+        assert self._export_mode is not None, "Please use a concrete instance of ExecutionManager"
+        try:
+            with torch.no_grad():
+                required_export_kwargs = {
+                    "input_names": self._input_info.names,
+                    "output_names": output_names,
+                    "opset_version": ortmodule.ONNX_OPSET_VERSION,
```

Comment:
onnx opset version can be independent from what ortmodule supports.

Reply:
Noted. PR updated.